app/TestTCP/testenv.hpp

```diff
@@ -183,6 +182,17 @@ class TestEnv2 : public ::testing::Test
 		host2->getPort(1)->setPropagationDelay(TimeUtil::makeTime(1, TimeUtil::MSEC));
 
 		switchingHub = new Switch("Switch1", &netSystem, Unreliable);
+
+		/*
+		 *  In order to make random components controllable.
+		 *  Editor : payload@kaist.ac.kr
+		 */
+		time_t randSeed = time(0);
+		if(char * inSeed = getenv("TCP_SEED"))
+			randSeed = atoi(inSeed);
+		printf("\n[TCP_SEED : %ld]\n", randSeed);
+		switchingHub->setRandSeed(randSeed);
+
```

Comment:
https://github.com/ANLAB-KAIST/KENSv3/blob/6fac040bca6526d117615d2b3b0a817b405a635e/app/TestTCP/testtransfer.cpp#L93

Please rename env var.

Please remove print statements.

Reply:
@leeopop 

For env var naming, `RANDOM_SEED` may occur unintended behavior. As I understood, `RANDOM_SEED` is used for generating random message data, not simulating unreliable things. Also, testransfer.cpp:93's seed doesn't have default value. Thus, when first run without environs, it won't be initialized with timestamp, but when `RANDOM_SEED` is given, it'll be initialized, therefore there will be contradiction. I think synchronizing two seed must be processed first. I think the simplest way to resolve this issue, I think call `serena` in testenv.hpp to set `RANDOM_SEED` when seed isn't given. I'll make a commit about this.

For print statements, we must have print statement to let student to know which seed is used. Since this commit's goal is reproducing a bug, KENS must tell what value is used for random seed. If you have better opinion or any reason to remove print statement, please tell me.

Reply:
1. If random seed must be printed, it must be done at the main function.
1. You may invoke `srand` and let other `RandomDistribution` instances to take `rand()` instead of `time(0)`. The single `srand` call will control all other random distributions.

Reply:
Got it. Since I'm out for now, I'll commit after going back. It might be after midnight, is it okay?